includes/acl/core/compressed_clip.h

```diff
@@ -84,4 +86,75 @@ namespace acl
 
 	static_assert(alignof(CompressedClip) == 16, "Invalid alignment for CompressedClip");
 	static_assert(sizeof(CompressedClip) == 16, "Invalid size for CompressedClip");
+
+	struct SegmentHeader
+	{
+		uint32_t				num_samples;
+		uint32_t				animated_pose_bit_size;						// TODO: Calculate from bitsets and formats?
+
+																			// TODO: Only need one offset, calculate the others from the information we have?
+		PtrOffset32<uint8_t>	format_per_track_data_offset;				// TODO: Make this offset optional? Only present if variable
+		PtrOffset32<uint8_t>	range_data_offset;							// TODO: Make this offset optional? Only present if normalized
+		PtrOffset32<uint8_t>	track_data_offset;
+	};
+
+	struct ClipHeader
+	{
+		uint16_t				num_bones;
+		uint16_t				num_segments;
+
+		RotationFormat8			rotation_format;
+		VectorFormat8			translation_format;
+		VectorFormat8			scale_format;								// TODO: Make this optional?
+
+		RangeReductionFlags8	clip_range_reduction;
+		RangeReductionFlags8	segment_range_reduction;
+
+		uint8_t					has_scale;
+
+		uint32_t				num_samples;
+		uint32_t				sample_rate;								// TODO: Store duration as float instead
+
+		PtrOffset16<SegmentHeader>	segment_headers_offset;
```

Comment:
In the long term I want to change all of this to use a more flexible format, something like optional parts of the header in strict absolute ordering for quick indexing

Reply:
Yeah, I remember - just wanted to do something in the meantime to avoid massive code duplication...